module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,65 @@
+#include "compression.hpp"
+
+#include <algorithm>
+#include <array>
+#include <iostream>
+#include <utility>
+#include <vector>
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(std::array<std::array<uint8_t, width>, height>& bitmap) {
+    std::vector<std::pair<uint8_t, uint8_t>> compressedData;
+    compressedData.reserve(width * height);
+
+    for (const auto& row : bitmap) {
+        if (row.empty())
+            continue;
+
+        uint8_t prevElem = row.front();
+        uint8_t cnt = 1;
+
+        for (auto it = row.begin(); it != row.end(); it++) {
+            if (*it == prevElem && it != row.begin()) {
+                cnt++;
+            }
+            if (*it != prevElem || it == std::prev(row.end())) {
+                compressedData.emplace_back(std::make_pair(prevElem, cnt));
```

Comment:
you don't need std::make_pair, you can make pair just by using {} notation

Reply:
It seems there is no big difference between std::make_pair and {} constructor for a pair during emplacing at back. However according to [main answer in this stack overflow thread](https://stackoverflow.com/questions/53900865/emplace-back-vs-push-back-when-inserting-a-pair-into-stdvector), it is good to use this syntax:
**vector.emplace_back( prevElem, cnt )**

Reply:
Thx